src/RabbitConnection/index.ts

```diff
@@ -23,7 +23,7 @@ export default class RabbitConnection {
    */
   private readonly $hostname: string
 
-  private readonly $vhost: string
+  private readonly $vhost: string | ""
```

Comment:
I think this don't need to be a property, I think it would be better if we just passed it directly to `handleHostname` method.

```ts
this.$hostname = handleHostname(
      this.rabbitConfig.hostname,
      this.rabbitConfig.vhost,
      this.rabbitConfig.port
)
```

And in `handleHostname` something like this

```
return port ? `${hostname}:${port}/${vhost || ''}` : hostname
```